src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -604,6 +604,30 @@ internal static bool IsRunningOnProcessorArchitectureARM()
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        internal static string GetTemporaryDirectory()
+        {
+            string tempDir = String.Empty;
+            string tempPath = Path.GetTempPath();
+            do
+            {
+                tempDir = Path.Combine(tempPath,System.Guid.NewGuid().ToString());
```

Comment:
If this is a common scenario, then perhaps we should require to specify a temporary directory using one of the possible ways?